### PR TITLE
Add bbram driver for ENE KB1200

### PR DESCRIPTION
**Pull request by ene-steven**
Add bbram driver for ENE KB1200

### CODE REVIEW

**Review by yperess — DISMISSED** [review]

File: dts/arm/ene/kb1200.dtsi [hunk 1]
```diff
@@ -335,6 +335,40 @@
 			#size-cells = <0>;
 			status = "disabled";
 		};
+
+		bbram: bbram@40041000 {
+			compatible = "ene,kb1200-bbram";
+			reg = <0x40041000 0x84>;
+			bbram-size = <124>;
```
Comment (decsny) [inline dts/arm/ene/kb1200.dtsi:342]: what is the difference between `bbram-size` and the size cell of the `reg` property here? I see the value is slightly different but similar

File: dts/bindings/memory-controllers/ene,kb1200-bbram.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+compatible: "ene,kb1200-bbram"
+
+include: base.yaml
+
+properties:
+  reg:
+    required: true
+
+  bbram-size:
```
Comment (decsny) [inline dts/bindings/memory-controllers/ene,kb1200-bbram.yaml:14]: this needs a description

**Review by decsny — CHANGES_REQUESTED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]
Please split this into multiple PRs, each adding one driver.  

This keeps the PR size manageable. Note that each driver subsystem in Zephyr has separate maintainers, so putting all of this into one big PR will make it really difficult to get it reviewed and approved.

**Comment by decsny** [comment]
also, how is this going to be tested, I don't see any twister yaml changes

File: boards/ene/kb1200_evb/kb1200_evb.yaml [hunk 3]
```diff
@@ -7,7 +7,13 @@ toolchain:
   - gnuarmemb
   - xtools
 supported:
+  - adc
   - gpio
+  - i2c
+  - pwm
+  - tach
```
Comment (decsny) [inline boards/ene/kb1200_evb/kb1200_evb.yaml:14]: I don't think "tach" is a real twister tag

**Review by decsny — DISMISSED** [review]

File: drivers/bbram/bbram_ene_kb1200.c [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	.check_power = bbram_kb1200_check_power,
+	.get_size = bbram_kb1200_get_size,
+	.read = bbram_kb1200_read,
+	.write = bbram_kb1200_write,
+};
+
+#define BBRAM_KB1200_DEVICE(inst)                                                                  \
+	static struct bbram_kb1200_data bbram_data_##inst = {};                                    \
+	static const struct bbram_kb1200_config bbram_cfg_##inst = {                               \
+		.vbat_ptr = (struct vbat_regs *)DT_INST_REG_ADDR(inst),                            \
```
Comment (yperess) [inline drivers/bbram/bbram_ene_kb1200.c:110]: Is the size not configurable?

**Review by yperess — CHANGES_REQUESTED** [review]

File: drivers/bbram/bbram_ene_kb1200.c (hunk 4 above)
Comment (ene-steven) [reply drivers/bbram/bbram_ene_kb1200.c:110]: Yes, the size is fixed.

**Pull request closed without merging**